docs/DeveloperGuide.md

```diff
@@ -1,15 +1,241 @@
 # Developer Guide
+This developer guide will help you to get started with our product, badMaths!
 
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements) 
+2. [Design & implementation](#design--implementation)
+   - [Store Notes](#store-notes) 
+   - [Graph](#graph)
+   - [Matrix](#matrix)
+2. [Product Scope](#product-scope)
+   - [Target user profile](#target-user-profile)
+   - [Value proposition](#value-proposition)
+3. [User Stories](#user-stories)
+4. [Non-Functional Requirements](#non-functional-requirements)
+5. [Glossary](#glossary)
+6. [Instructions for manual testing](#instructions-for-manual-testing)
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+
 ## Design & implementation
```

Comment:
Concise and clear explanation of what your features does in this section! Consider explaining why you implemented your feature logic this way, maybe even describing some alternative implementations you considered but ultimately dropped, which would strengthen the appeal of your features to a developer.